### PR TITLE
Refactoring backend [Work In Progress]

### PR DESCRIPTION
**Pull request by italo-batista**
Sorry, it is a huge PR. 

I ask to each of you to take one modified controller and test if its behavior is still the same of if there is some error. I cant test all controllers because there are many ones and there is still a lot of refactoring work to do.

You can comment when you test, so another person won't test the same controller.

**CONTROLLERS**
- [x] Watch
- [X] Rated
- [X] Follows
- [X] FollowsPage
- [X] Action
- [X] Person
- [x] TvShow
- [x] Rated
- [x] Season
- [x] Episode

@oandrevictor  @DavidMedeiros @gabriela-motta @gabrielmla @Rafaela9136

### CODE REVIEW

**Comment by italo-batista** [comment]
There are some still needed modifications but it is already possible to test.

Please take a time to test some API endpoints. Dont forget to say which controller you tested, so I can keep track of the list above.

 This PR is very important and needed. And some tasks of the current sprint depends on this modifications.

File: server/controllers/watchedController.js [hunk 1]
```diff
@@ -70,120 +67,63 @@ exports.update = async function(req, res) {
...
     https.get("https://api.themoviedb.org/3/tv/"+ tv_id + "/season/"+ season_number +"?api_key=db00a671b1c278cd4fa362827dd02620",
     (resp) => {
       let data = '';
       resp.on('data', (chunk) => {
         data += chunk;
       });
       resp.on('end', () => {
-        console.log("saving season result to redis:"+  query)
+        console.log("saving season result to redis:"+  query);
         client.set(query, JSON.stringify(data));
```
Comment (gabrielmla) [inline server/controllers/watchedController.js]: Client not defined,

File: server/controllers/watchedController.js [hunk 2]
```diff
@@ -70,120 +67,63 @@ exports.update = async function(req, res) {
...
-      var value = await getSeasonFromAPI(media_obj._tmdb_tvshow_id, media_obj.season_number).then((season)=>{
-        var watched_with_full_media = watched_obj;
-        watched_with_full_media._media = media_obj
+};
+
+var injectMediaJsonInWatched = async function(watched_obj) {
+    let mediaId = watched_obj._media;
+    let mediaObj = await DataStoreUtils.getMediaObjById(mediaId);
+    if (mediaObj.__t == 'Episode' && mediaObj._tmdb_tvshow_id){
+      var value = await getSeasonFromAPI(media_obj._tmdb_tvshow_id, media_obj.season_number).then((season) => {
```
Comment (gabrielmla) [inline server/controllers/watchedController.js]: Mudar media_obj para MediaObj

File: server/controllers/watchedController.js [hunk 3]
```diff
@@ -70,120 +67,63 @@ exports.update = async function(req, res) {
...
+        var watched_with_full_media = watchedObj;
```
Comment (gabrielmla) [inline server/controllers/watchedController.js]: A função recebe como parametro watched_obj, mas aqui é usado watchedObj causando erro.

File: server/controllers/watchedController.js (hunk 2 above)
Comment (italo-batista) [reply server/controllers/watchedController.js]: Thank you. Resolved

File: server/controllers/watchedController.js (hunk 3 above)
Comment (italo-batista) [reply server/controllers/watchedController.js]: Thank you. Resolved

File: server/controllers/watchedController.js (hunk 1 above)
Comment (italo-batista) [reply server/controllers/watchedController.js]: Esse método precisa ser refatorado. Ele foi copiado de tvShow, eu acho. Sendo que tvShow tem um client e watched. Se tu precisar agr, faz RedisClient.createAndAuth.

Comment (gabrielmla) [reply server/controllers/watchedController.js]: Coloquei o RedisClient.createAndAuth e pegou. Mas faz sentido esse método ser refatorado pois ele é doido, mas eu não sei nem pra onde essas coisas do cache/api tão indo então não sei fazer isso.

**Comment by oandrevictor** [comment]
Não tá carregando as paginas que uma pessoa segue

**Pull request merged**